allennlp/modules/transformer/bert_biattention.py

```
@@ -0,0 +1,122 @@
+import math
+
+import torch
+
+from allennlp.common import FromParams
+
+
+class BertBiAttention(torch.nn.Module, FromParams):
+    def __init__(
+        self,
+        hidden_size1: int,
+        hidden_size2: int,
+        combined_hidden_size: int,
+        num_attention_heads: int,
+        dropout1: float,
+        dropout2: float,
+    ):
+        super().__init__()
+        if combined_hidden_size % num_attention_heads != 0:
+            raise ValueError(
+                "The hidden size (%d) is not a multiple of the number of attention "
+                "heads (%d)" % (combined_hidden_size, num_attention_heads)
+            )
+
+        self.num_attention_heads = num_attention_heads
+        self.attention_head_size = int(combined_hidden_size / num_attention_heads)
```

We should throw an exception if this is not an even division.